benchmarks/benchmarks/stats.py

```diff
@@ -57,45 +57,107 @@ def time_ttest_ind_diff_var(self):
+            # picking arbitrary value for this
+            self.args = [.99, *args[1:]]
+            properties = ('pmf' if isinstance(self.dist, stats.rv_discrete)
+                          else 'pdf')
+        elif properties == 'logpdf/logpmf':
+            properties = ('logpmf' if isinstance(self.dist, stats.rv_discrete)
+                          else 'logpdf')
+            self.args = args
+        elif properties == 'isf':
+            self.args = [.99, *args[1:]]
+        elif properties == 'moment':
+            self.args = [5, *args[1:]]
+        elif properties.startswith('stats_'):
+            properties = 'stats'
+            self.args = args[1:]
+            kwds['moments'] = properties[6:]
+            self.kwds = kwds
+        elif properties in ['entropy']:
+            self.args = args[1:]
+        elif properties == 'expect':
+            self.args = []
+            kwds['args'] = tuple(args[1:])
+        elif properties == 'interval':
+            self.args = [.99, *args[1:]]
+        else:
+            self.args = args
+        self.kwds = kwds
+
+        try:
+            self.method = getattr(self.dist, properties)
+        except AttributeError:
+            raise NotImplementedError("This attribute is not a member "
+                                      "of the distribution")
 
     def time_distribution(self, distribution, properties):
-        if distribution == 'gamma':
-            if properties == 'pdf':
-                stats.gamma.pdf(self.x, a=5, loc=4, scale=10)
-            elif properties == 'cdf':
-                stats.gamma.cdf(self.x, a=5, loc=4, scale=10)
-            elif properties == 'rvs':
-                stats.gamma.rvs(size=1000, a=5, loc=4, scale=10)
-            elif properties == 'fit':
-                stats.gamma.fit(self.x, loc=4, scale=10)
-        elif distribution == 'cauchy':
-            if properties == 'pdf':
-                stats.cauchy.pdf(self.x, loc=4, scale=10)
-            elif properties == 'cdf':
-                stats.cauchy.cdf(self.x, loc=4, scale=10)
-            elif properties == 'rvs':
-                stats.cauchy.rvs(size=1000, loc=4, scale=10)
-            elif properties == 'fit':
-                stats.cauchy.fit(self.x, loc=4, scale=10)
-        elif distribution == 'beta':
-            if properties == 'pdf':
-                stats.beta.pdf(self.x, a=5, b=3, loc=4, scale=10)
-            elif properties == 'cdf':
-                stats.beta.cdf(self.x, a=5, b=3, loc=4, scale=10)
-            elif properties == 'rvs':
-                stats.beta.rvs(size=1000, a=5, b=3, loc=4, scale=10)
-            elif properties == 'fit':
-                stats.beta.fit(self.x, loc=4, scale=10)
+        self.method(*self.args, **self.kwds)
 
+
+class Distribution_old(Benchmark):
+    param_names = ['distribution', 'properties']
+    params = [[], []]
+
+    def setup(self):
+        pass
+
+    def time_distribution(self, distribution, properties):
+        pass
```

Comment:
I assume you've done this because it's required to preserve the old results? How does it work?

Reply:
keeping old benchmark class unmodified so it is retained